lib/index.js

```diff
@@ -294,33 +286,61 @@ function convertCallbackFunction(V, opts) {
 
 const abByteLengthGetter =
     Object.getOwnPropertyDescriptor(ArrayBuffer.prototype, "byteLength").get;
+const sabByteLengthGetter =
+    Object.getOwnPropertyDescriptor(SharedArrayBuffer.prototype, "byteLength").get;
 
 function isArrayBuffer(V) {
     try {
+        // This will throw on SharedArrayBuffers, but not detached ArrayBuffers.
+        // (The spec says it should throw, but the spec conflicts with implementations: https://github.com/tc39/ecma262/issues/678)
         abByteLengthGetter.call(V);
+
+        // This will throw on detached ArrayBuffers, but will not throw on SharedArrayBuffers.
+        // eslint-disable-next-line no-new
+        new Uint8Array(V);
         return true;
-    } catch (e) {
+    } catch {
+        return false;
+    }
+}
+
+function isSharedArrayBuffer(V) {
+    try {
+        sabByteLengthGetter.call(V);
+        return true;
+    } catch {
         return false;
     }
 }
 
-// I don't think we can reliably detect detached ArrayBuffers.
-exports.ArrayBuffer = (V, opts) => {
+function isNonSharedArrayBufferView(V) {
+    return ArrayBuffer.isView(V) && isArrayBuffer(V.buffer);
```

Comment:
How reliable do we want to get, in case the user overrode `V.buffer`, or `TypedArray.prototype.buffer`?

Reply:
My fuzzy bar so far has been that jsdom does not need to be resilient in the face of monkeypatching globals/prototypes, but it should be resilient in the face of arbitrary objects as inputs. So overriding `V.buffer` is a problem, by that definition, hmm.